keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -1274,15 +1277,63 @@ that has to be considered and addressed.
+to "now".  Given that watch requests may be called with potentially quite
+old resource version, we may need to go over many events, filter them out
+and send the ones that are matching selectors.
+
+We will solve this problem by also handling watch requests by our priority
+and fairness kube-apiserver filter.  The queueing and admitting of watch
+requests will be happening exactly the same as for all non-longrunning
+requests.  However, as soon as watch is initialized, it will be sending
+an articial `finished` signal to the APF dispatcher - after receiving this
+signal dispatcher will be treating the request as already finished (i.e.
+the concurrency units it was occupying will be released and new requests
+may potentially be immediately admitted), even though the request itself
+will be still running.
+
+However, there are still some aspects that require more detail discussion.
+
+##### Getting the initialization signal
+
+The first question to answer is how we will know that watch initialization
+has actually been done.  However, the answer for this question is different
+depending on whether the watchcache is on or off.
+
+In watchcache, the initialization phase is clearly separated - we explicily
+compute `init events` and process them.  What we don't control at this level
+is the process of serialization and sending out the events.
+In the initial version we will ignore this and simply send the `initialization
+done` signal as soon as all the init events are in the result channel.  In the
+future, we can estimate it by assuming some constant (or proportional to
+event size) time for an event and delaing delivering the signal by the sum of
+those across all init events.
+
+If watchcache is disabled, the watch is just proxied to the etcd.  In this
+case we don't have an easy way to answer the question whether the watch has
+already catch up.  Given that watchcache is enabled by default for all
+resources except from events, in the initial version we will simply deliver
+the `initialization done` signal as soon as etcd watch is started.  We will
+explore how to improve it in the future.
+
+##### Passing the initialization signal to the filter
+
+The second aspect is passing the `initialization done` signal from watchcache
+or etcd3 storage layer to the priority and fairness filter.  We will do that
+using a synchronization primitives (either channel or WaitGroup) and putting
+that into request context.  Given that context is passed down with the request,
+watchcache/etcd3 storage will have access to it and will be able to pass the
+signal.
+
+##### Width of the request
+
+The performance characteristics of the watch initialization is very different
+than processing a regular single-object request, we should (similarly as for
+LIST requests) adjust the `width` of the request.  However, in the initial
+version, we will just use `width=1` for all watch requests.  We will decide
```

Comment:
In that future state, I'd expect something similar to LIST width calculations.  Doesn't need to be described here, just indicating where my head is at.

Reply:
The next sentence was supposed to reflect that - rephrased to make it more explicit.